src/app/(app)/course/[courseId]/project-sets/[projectSetId]/(hooks)/useProjects.tsx

```diff
@@ -0,0 +1,68 @@
+'use client'
+
+import {createContext, useContext, useEffect, useState, type PropsWithChildren, type FC} from "react"
+import {useParams} from "next/navigation"
+import {type Project} from "@/_temp_types/projects"
+import {toast} from "@/hooks/use-toast"
+
+type ProjectSetContextType = {
+  projectSetId: string
+  projects: Project[]
+  displayProjects: Project[]
+  setDisplayProjects: (projects: Project[]) => void
+  currentProject: Project | undefined
+  setCurrentProject: (project: Project) => void
+}
+
+const ProjectSetContext = createContext<ProjectSetContextType>({
+  projectSetId: "",
+  projects: [],
+  displayProjects: [],
```

Comment:
still reviewing but have my doubts if we need the logic of `displayProjects` to exist at the provider level. I kinda think the left sidebar could handle this logic without anything else under the context needing to be aware of it